src/payment/data/service.js

```diff
@@ -114,3 +114,13 @@ export async function deleteCoupon(id) {
     .catch(handleBasketApiError);
   return transformResults(data);
 }
+
+export async function checkBasketForDiscount(courseKey) {
```

Comment:
in ecommerce (and mfe) basket loads, async request goes to lms to check if a discount should be  applied, if it is true, basket is recreated server side and page reloads (or re-renders in the case of the mfe)

Reply:
Nit: we've been trying to prefix these service method names with an HTTP verb/have their name represent the endpoint, for what it's worth.  Something like `getCourseDiscount`, maybe?

Reply:
updated to getDiscountData